sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/Filter.cs

```diff
@@ -513,32 +528,32 @@ private Expression ProduceComparatorExpressionForSingleFieldCondition(Expression
                                     ThrowOnInvalidFilter(fieldType, !comparandDouble.HasValue);
                                     return Expression.Equal(
                                         fieldConvertedExpression,
-                                        Expression.Constant(comparandDouble.Value, isFieldTypeNullable ? typeof(Nullable<>).MakeGenericType(typeof(double)) : typeof(double)));
```

Comment:
The `MakeGenericType` here isn't trimmable. It's also not necessary. 

Looking at the second half of the statement, we already know the type is a `double` and we use `typeof(double)`. We can replace the `MakeGenericType` with a `typeof(double?)`